pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -323,11 +323,12 @@ func (c *Cluster) handleExternalMonStatus(status client.MonStatusResponse) error
 
 func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (bool, error) {
 	var changed bool
-	// Populate a map with the current monitor from ClusterInfo
-	monsNotFound := map[string]interface{}{}
-	for _, mon := range c.ClusterInfo.Monitors {
-		monsNotFound[mon.Name] = struct{}{}
+	// clearing the content of clusterinfo monitors, it'll get populated in the next iteration
+	for mon := range c.ClusterInfo.Monitors {
+		delete(c.ClusterInfo.Monitors, mon)
```

Comment:
I wonder if `c.ClusterInfo.Monitors = {}` could be enough here to clear instead of iterating.
Do you mind giving it a try?

Reply:
sure, let me make these changes and try

Reply:
I think that's not possible 
I followed this https://stackoverflow.com/questions/13812121/how-to-clear-a-map-in-go